x/evm/module.go

```diff
@@ -39,8 +41,10 @@ import (
 )
 
 var (
-	_ module.AppModule      = AppModule{}
-	_ module.AppModuleBasic = AppModuleBasic{}
+	_ module.AppModule          = AppModule{}
+	_ module.AppModuleBasic     = AppModuleBasic{}
+	_ appmodule.HasEndBlocker   = AppModule{}
+	_ appmodule.HasBeginBlocker = AppModule{}
 )
```

Comment:
Add 3 interface
```
_ module.HasGenesis          = AppModule{}
_ module.HasConsensusVersion = AppModuleBasic{}
_ module.HasServices         = AppModule{}
```

And we can remove unused func.
RegisterInvariants, RegisterStoreDecoder, GenerateGenesisState, WeightedOperations

Reply:
https://github.com/xpladev/ethermint/commit/3f68bb88b9a78e7d3b3eca175997bb81f29fc03d
https://github.com/xpladev/ethermint/commit/0dd3e83b3d83abc20727e63bc0dcb5f5c3703dde